server/src/main/java/org/elasticsearch/search/retriever/rankdoc/RankDocsQuery.java

```diff
@@ -164,12 +166,8 @@ public float getMaxScore(int docId) {
                         }
 
                         @Override
-                        public float score() {
-                            // We could still end up with a valid 0 score for a RankDoc
-                            // so here we want to differentiate between this and all the tailQuery matches
-                            // that would also produce a 0 score due to filtering, by setting the score to `Float.MIN_VALUE` instead for
-                            // RankDoc matches.
```

Comment:
Please preserve the comment (maybe moving it to the ctr) to indicate that the minimum score should be strictly greater than 0. I also wonder if the two should be decorrelated. It should be possible to set the minimum score to be `0` (to remove documents that only match filters) but in that case we still need to replace 0 with ` Float.MIN_VALUE)` for the reason invoked in the comment.